public/scripts/utils.js

```diff
@@ -17,3 +17,11 @@ export function $try(func) {
 export function reformatDatetime(dtStr) {
   return new Date(dtStr).toString();
 }
+
+export function downloadCSV(data, filename) {
```

Comment:
If you have an endpoint that serves up the CSV, then you can just have an `<a href=...>` tag that points straight there, if I'm not missing anything more subtle.